### ISSUE TITLE
Add toEither method to Try that supplies Throwable

### ISSUE THREAD

**Issue by jlorenzen** [vavr-io/vavr#2291]
What do you guys think about adding a `toEither` method to `Try` that supplies the `Throwable`? Seems like I'm always doing this but it currently requires two steps:
```java
Try.of(() -> invoke())
    .toEither()
    .mapLeft(t -> mapToError(t)
```

For something that seems so common, I think it would be nice if we could do:
```java
Try.of(() -> invoke())
    .toEither(t -> mapToError(t))
```

Thoughts?

**Comment by danieldietrich** [comment]
Hi, this is generally a good idea. In fact I would favor toEither(Function) over toEither() in order to keep the API surface area small.
However, when having fine-grained Java modules, a direct conversion method on the Try type isn‘t possible anymore. Try does not Either. It might happen that these converson methods will be moved to API, e.g.

```java
import static io.vavr.api.*;

toEither(myTry, mapError);
```

That would be the price of fine-grained modularization.

However, I‘m still thinking about putting Either, Try and Option into the same module. Then your suggestion would be possible. Maybe that is the better choice.

**Comment by jlorenzen** [comment]
@danieldietrich Yeah I'm not sure of the overall direction for vavr 1.x. I know you've been planning some pretty major changes. I know I wouldn't necessarily be a fan if I had to do this:
```java
toEither(Try.of(() -> invoke()), t -> mapError(t))
```

It quickly becomes less readable I think as you add more stuff to the Try (ie map, flatMap, filter, etc).

If you want, I'd love to submit a PR to add `Try#toEither(Function<? super L, ? extends U> leftMapper)`. Just let me know.

**Comment by danieldietrich** [comment]
@jlorenzen Please hold your horses. I‘m currently setting up the first version of vavr-control-1.0. I will add the conversion methods today.

> Yeah I'm not sure of the overall direction for vavr 1.x.

I‘m sure you will be satisfied. The conversion methods will not go to API...

**Comment by CauchyPeano** [comment]
This one also intersecting with similar issue https://github.com/vavr-io/vavr/issues/2207

**Comment by sclassen** [comment]
Will this be backported for people stuck on vavr 0.9 ??

**Comment by danieldietrich** [comment]
@sclassen wrt semantic versioning, additional features would be released in a 0.10 instead (which isn‘t planned currently). 0.9 will receive fixes only.

**Comment by danieldietrich** [comment]
Vavr 1.0.0 contains a final set of conversion methods. See master branch. A `vavr-collection-1.0.0-alpha-1` will be released soon.
